### PR TITLE
DENG-683 Updated bq usage table and created new cost view

### PR DESCRIPTION
**Pull request by wwyc**
[DENG-683](https://mozilla-hub.atlassian.net/browse/DENG-683):  

*  Updates were made to increase alignment with the [billing.bigquery_usage_by_user](https://github.com/mozilla-services/dataops/blob/main/bigquery/billing/billing.bigquery_usage_by_user.sql) table since it has been used by the platform team to monitor costs. 

* New columns were added to allow more filtering and additional query information
* New view is created to calculate costs

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

[DENG-683]: https://mozilla-hub.atlassian.net/browse/DENG-683?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by gkabbz — APPROVED** [review]

File: sql/moz-fx-data-shared-prod/monitoring/bigquery_usage_costs/view.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      source_project,
+      user_email,
+      job_id,
+      total_terabytes_processed
+    FROM
+      `moz-fx-data-shared-prod.monitoring.bigquery_usage`
+    WHERE
+      state = 'DONE'
+      AND job_type = 'QUERY'
+      AND reservation_id IS NULL
```
Comment (lucia-vargas-a) [inline sql/moz-fx-data-shared-prod/monitoring/bigquery_usage_costs/view.sql:21]: For context, what's the value of `reservation_id` if it's filtering only NULLs?  Do we want to exclude jobs related to certain processes e.g. `reservation_id` = default-pipeline?

File: sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v1/schema.yaml [hunk 2]
```diff
@@ -20,6 +20,16 @@ fields:
   type: STRING
   description: The type of the job
 
+- mode: NULLABLE
+  name: reservation_id
+  type: STRING
+  description: Name of the primary reservation (slot allocation) assigned to this job
```
Comment (lucia-vargas-a) [inline sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v1/schema.yaml:26]: Would this description provide a quick understanding for those who are not familiar with the concept of reservations? Feel free to adjust it,
```suggestion
  description: Name of the primary reservation assigned to this job. Reservations are BigQuery slots allocated to projects in order to distribute the resources and processing capacity.
```

Comment (wwyc) [reply sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v1/schema.yaml:26]: Thanks for the suggestion.  This makes it more clear.

File: sql/moz-fx-data-shared-prod/monitoring/bigquery_usage_costs/view.sql (hunk 1 above)
Comment (wwyc) [reply sql/moz-fx-data-shared-prod/monitoring/bigquery_usage_costs/view.sql:21]: I don't have enough context on reservations to know which filter is appropriate.  The ask was to mimic [billing.bigquery_usage_by_user](https://github.com/mozilla-services/dataops/blob/main/bigquery/billing/billing.bigquery_usage_by_user.sql) and use the same filtering.

**Pull request merged**